### DISCUSSION TITLE
Flexible schedule via Home Assistant

### DISCUSSION THREAD

**Discussion by JeanOlivier** [revk/xESP32-Faikout#132] [Show and tell]
The schedule included in the firmware is very convenient, but maybe like me you'd like to have more flexibility.
e.g. have different schedules for the weekends and week days, or more than one ON and OFF per day.

That's why I wrote a small python script that uses pyscript from HACS to control my Daikin.
I'm sharing here in the hope that it might be of use to others.

See https://hacs-pyscript.readthedocs.io/en/latest/installation.html to install and configure pyscript. Don't forget to reboot the system that Home Assistant is running on afterwards.

You can find my script here: https://gist.github.com/JeanOlivier/d69da57ec87129a8b10046ff558cd30c#file-daikin_scheduler-py

It should be fairly self explanatory if you understand python. Just configure the header according to your needs.

**Comment by BenJamesAndo** [comment]
What would the advantage of using this instead of just setting up an automation to turn the AC on and off within Home Assistant? Perhaps because this is one file, whereas in HA you might need multiple automations for different days etc.?

**Reply by JeanOlivier** [reply]
1. I know Python well, HA automations seem unintuitive to me.
2. It's one file, easy to edit on my phone using the file explorer plugin.
3. My goal was to be very flexible, eventually I want to turn on either the Daikin or electrical baseboards depending on the outside temperature.
4. It is a simple script I wrote to explore the pyscript integration before tackling a PID, also available in this discussion section, here: https://github.com/revk/ESP32-Faikin/discussions/177

**Comment by revk** [comment]
I'll have a look
